src/RequestHandler/BaseHandlerTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace SilverStripe\MFA\RequestHandler;
+
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\MFA\Service\RegisteredMethodManager;
+use SilverStripe\MFA\Store\StoreInterface;
+use SilverStripe\Security\Member;
+use SilverStripe\View\Requirements;
+
+trait BaseHandlerTrait
+{
+    /**
+     * Perform the necessary "Requirements" calls to ensure client side scripts are available in the response
+     */
+    protected function applyRequirements(): void
+    {
+        // Run through requirements
+        Requirements::javascript('silverstripe/mfa: client/dist/js/injector.js');
+        Requirements::javascript('silverstripe/admin: client/dist/js/i18n.js');
+        Requirements::javascript('silverstripe/mfa: client/dist/js/bundle.js');
+        Requirements::add_i18n_javascript('silverstripe/mfa: client/lang');
+        Requirements::css('silverstripe/mfa: client/dist/styles/bundle.css');
+
+        // Plugin module requirements
+        foreach (MethodRegistry::singleton()->getMethods() as $method) {
+            $method->applyRequirements();
+        }
+    }
+
+    /**
+     * @return StoreInterface|null
+     */
+    protected function getStore(): ?StoreInterface
+    {
+        if (!$this->store) {
+            $spec = Injector::inst()->getServiceSpec(StoreInterface::class);
+            $class = is_string($spec) ? $spec : $spec['class'];
+            $this->store = call_user_func([$class, 'load'], $this->getRequest());
```

Comment:
I know we've talked about this many times already, but I'm still really not convinced that we need to be able to construct a Store from a Member AND to "load" (construct) a Store from an HTTPRequest as well

Reply:
Yeah my response is always the same. There are two ways we get a store. One when we have a member and we're initialising the store for the first time and the other is when we're loading the store back (in `SessionStore`'s case - from session).

I think there's probably some way we could make this "more correct" by created some factory class or something but I didn't really want to delve into that for this PR. Right now this just copies the logic from the previous `LoginHandler`.

Reply:
Yeah maybe remove the constructor and have `loadFromRequest()` and `loadFromMember()` methods. Can think more about it and address it the next time I moan about it =D